### PR TITLE
Add SQL Server 2022 workflow contained on runner VM with Python

### PR DESCRIPTION
**Pull request by aaronburtle**
This PR adds a new .yaml file which will handle the transition of SQL2022 ML Utils from utilizing Azure Resources by providing support for Python. This aligns SQLMLUtils for SQL Server 2022 with the first phase of the security push.

To demonstrate that the SQL Server 2022 environment is setup correctly we have added a `sp_execute_external_script` task for Python as well, so that we can validate that the GHA is capable of correctly executing an external script in R using the machine learning services.

Link to successful pipeline run: https://github.com/microsoft/sqlmlutils/actions/runs/9898339030/job/27344914128?pr=118

### CODE REVIEW

File: .github/workflows/SQL2022.yml [hunk 1]
```diff
@@ -154,3 +181,28 @@ jobs:
         env:
           PASSWORD_AIRLINE_USER: "${{ env.dbPassword }}"
           PASSWORD_AIRLINE_USER_DBOWNER: "${{ env.dbPassword }}"
+
+      - name: Set up Python ${{ env.python-version }}
+        uses: actions/setup-python@v2
```
Comment (seantleonard) [inline .github/workflows/SQL2022.yml:186]: nit: `actions/setup-python` is now on version 5. Can you check whether upgrading has any negative effects?

File: .github/workflows/SQL2022.yml [hunk 2]
```diff
@@ -154,3 +181,28 @@ jobs:
...
+        with:
+          python-version: ${{ env.python-version }}
+
+      - name: Install Python dependencies
+        working-directory: ./Python
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install flake8 pytest
+          pip install -r requirements.txt
```
Comment (seantleonard) [inline .github/workflows/SQL2022.yml:195]: link to pipeline run (successful?) in pr description.

File: .github/workflows/SQL2022.yml [hunk 3]
```diff
@@ -110,6 +110,27 @@ jobs:
       - name: Configure the R runtime installed for MLS with SQL Server
         run: C:\MLS\R\library\RevoScaleR\rxLibs\x64\RegisterRext.exe /configure /rhome:"C:\MLS\R" /instance:"MSSQLSERVER"
 
+      - name: Install Python for MLS
```
Comment (seantleonard) [inline .github/workflows/SQL2022.yml:113]: in a yaml comment above this section, Link to the Python installation and configuration section of the SQL 22 MLS install docs. Make sure to remove the `en-us` from the URL.

**Review by seantleonard — CHANGES_REQUESTED** [review]
pending pipeline run before merge.

File: .github/workflows/SQL2022.yml [hunk 4]
```diff
@@ -131,6 +154,12 @@ jobs:
           @input_data_1 =N'SELECT 1 AS hello'
           WITH RESULT SETS (([hello] int not null));"
 
+      - name: Execute SPEES for Python
```
Comment (Aniruddh25) [inline .github/workflows/SQL2022.yml:157]: nit: SPEES is not a well known acronym. Please use `sp_execute_external_script`

File: .github/workflows/SQL2022.yml [hunk 5]
```diff
@@ -131,6 +154,12 @@ jobs:
...
+        run: sqlcmd -S localhost -U SA -P %dbPassword% -l 5 -Q "
+          EXEC sp_execute_external_script  @language =N'Python',
+          @script=N'OutputDataSet = InputDataSet;',
+          @input_data_1 =N'SELECT 1 AS hello' WITH RESULT SETS (([hello] int not null));"
```
Comment (Aniruddh25) [inline .github/workflows/SQL2022.yml:161]: do we fail the pipeline if this step fails?

**Review by Aniruddh25 — APPROVED** [review]
Few suggestions, otherwise looks good!

File: .github/workflows/SQL2022.yml (hunk 5 above)
Comment (aaronburtle) [reply .github/workflows/SQL2022.yml:172]: It will fail if there is an error.

**Review by seantleonard — APPROVED** [review]

**Pull request merged**